src/transformers/generation_stopping_criteria.py

```diff
@@ -83,15 +83,16 @@ class StoppingCriteriaList(list):
     def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs) -> bool:
         return any(criteria(input_ids, scores) for criteria in self)
 
+    @property
+    def max_length(self) -> Optional[int]:
+        for stopping_criterium in self:
+            if isinstance(stopping_criterium, MaxLengthCriteria):
+                return stopping_criterium.max_length
+
 
 def validate_stopping_criteria(stopping_criteria: StoppingCriteriaList, max_length: int):
-    found = False
-    for stopping_criterium in stopping_criteria:
-        if isinstance(stopping_criterium, MaxLengthCriteria):
-            found = True
-            if stopping_criterium.max_length != max_length:
-                warnings.warn(
-                    "You set different `max_length` for stopping criteria and `max_length` parameter", UserWarning
-                )
-    if not found:
+    smax_length = stopping_criteria.max_length
+    if smax_length is not None and smax_length != max_length:
+        warnings.warn("You set different `max_length` for stopping criteria and `max_length` parameter", UserWarning)
+    elif smax_length is None:
         stopping_criteria.append(MaxLengthCriteria(max_length=max_length))
```

Comment:
I think it's more in line with the libraries' philosophy if we could actually return the `stopping_criteria` here so that this line:

```python
validate_stopping_criteria(stopping_criteria, max_length)
```

becomes this line:
```python
stopping_criteria = validate_stopping_criteria(stopping_criteria, max_length)
```

Since the function `validate_stopping_criteria(...)` changes one of its argument, I think it's more understandable if the function returns the changed variable - otherwise one might assume that the function only checks something but doesn't actually change anything - what do you think?

Reply:
We can, but I feel like we should change the function to make it functional (recreate a new obejct) if I go that way.
It's super dangerous to have a function that looks functional but is not.

I switched to a functional function that returns a new stopping_criteria_list, it should be mostly very low cost, and much less surprising in behavior